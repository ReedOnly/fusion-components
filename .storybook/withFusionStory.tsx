import React from 'react';
import Header from '../src/components/core/Header';
import Content from '../src/components/core/Content';
import Overlay from '../src/components/general/ApplicationGuidance/components/Overlay';

export default (title: string, padding: number = 16) => (stories: () => React.ReactNode) => {
    return (
        <>
            <Overlay fixed>
                <Header aside={null} start={null} content={null} />
            </Overlay>
            <Content>
                <Overlay>
                    <div
                        style={{
                            padding,
                            minHeight: '100%',
                            width: '100%',
                            maxWidth: '100%',
                            boxSizing: 'border-box',
                        }}
                    >
                        {stories()}
                    </div>
                </Overlay>
            </Content>
        </>
    );
};
