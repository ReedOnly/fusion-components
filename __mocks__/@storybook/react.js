import renderer from 'react-test-renderer';
import addons, { mockChannel } from '@storybook/addons';
import { withFusionContext } from '../../.storybook/withFusionContext';

addons.setChannel(mockChannel());

export const action = (actionName) => jest.fn();
const fusionContext = withFusionContext();
export const storiesOf = (groupName) => {
    const api = {
        decorators: [],
        parameters: {},

        add(storyName, story) {
            if (api.parameters.skipTest) {
                return;
            }

            describe(groupName, () => {
                it(storyName, () => {
                    // const decoratedStory = api.decorators.reduce((s, decorator) => {
                    //     try {
                    //         return decorator(s, { parameters: api.parameters });
                    //     } catch (e) {
                    //         return s;
                    //     }
                    // }, story);

                    const component = renderer.create(fusionContext(story));

                    // @TODO Enable custom names once released:
                    // > https://github.com/facebook/jest/pull/2094
                    expect(component.toJSON())
                        .toMatchSnapshot
                        // `${groupName}.${storyName}`
                        ();
                });
            });

            return api;
        },

        addDecorator() {
            return api;
        },

        addParameters(parameters) {
            api.parameters = { ...api.parameters, ...parameters };
            return api;
        },
    };

    return api;
};

export const addDecorator = () => {};
