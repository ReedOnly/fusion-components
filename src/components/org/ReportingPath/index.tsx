import React from "react";


const ReportingPath = () => {
    return(
        <div>
r
        </div>
    )
}

export default ReportingPath;