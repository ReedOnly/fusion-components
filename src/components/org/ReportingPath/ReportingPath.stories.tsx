import React from 'react';
import { storiesOf } from '@storybook/react';
import withFusionStory from '../../../../.storybook/withFusionStory';
import ReportingPath from '.';
import { OrgStructure, OrgChartItemProps } from '@equinor/fusion-components';

type Position = OrgStructure & {
    name?: string;
    positionId?: string;
    personName?: string;
};
const positions: Position[] = [
    {
        id: '1',
        name: 'Leader',
        positionId: '123',
        personName: 'James Test',
    },
    {
        id: '2',
        parentId: '1',
        name: 'Accountant',
        positionId: '123',
        personName: 'Sam Test',
        aside: true,
    },
    {
        id: '5',
        parentId: '2',
        name: 'Worker',
        positionId: '3',
        personName: 'Sam Test',
        aside: true,
    },
];

const cardStyle = {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
    boxShadow:
        '0px 1px 5px rgba(0, 0, 0, 0.2), 0px 3px 4px rgba(0, 0, 0, 0.12), 0px 2px 4px rgba(0, 0, 0, 0.14)',
    margin: '8px',
    boxSizing: 'border-box',
    flex: '1',
    fontSize: '16px',
    height: 'calc(100% - 16px)',
} as React.CSSProperties;

const PositionCard: React.FC<OrgChartItemProps<Position>> = ({ item }) => {
    return (
        <div style={cardStyle}>
            Position : {item.name}
            <br />
            Person : {item.personName}
        </div>
    );
};

const ReportingPathStory = () => {
    return <ReportingPath structure={positions} component={PositionCard} />;
};

storiesOf('Org|Reporting Path', module)
    .addDecorator(withFusionStory('Reporting Path'))
    .add('Default', () => <ReportingPathStory />);
