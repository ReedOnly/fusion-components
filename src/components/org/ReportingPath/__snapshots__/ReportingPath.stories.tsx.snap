// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pro org|Reporting Path Default 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <svg
      height={0}
      viewBox="0 0 0 0"
      width={0}
    >
      <g
        className="links"
      />
      <g
        className="children"
      />
      <g
        className="children"
      />
    </svg>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;
